index.js

```diff
@@ -1,395 +1,157 @@
-  if (typeof type === 'string') {
-    switch (type) {
-      case 'string': return '""'
-      case 'number': return '0'
-      case 'boolean':
-      case 'bool': return 'false'
-      default: throw new Error('Not default type available')
-    }
-  }
-  return 'undefined'
-}
-
-function endNumberString (s, ptr) {
-  for (; ptr < s.length; ptr++) {
-    switch (s.charCodeAt(ptr)) {
-      case 44:
-      case 93:
-      case 125:
-      return ptr
-    }
-  }
-  throw new Error('Could not find end of number')
-}
-
-function endNumberBuffer (b, ptr) {
-  for (; ptr < b.length; ptr++) {
-    switch (b[ptr]) {
-      case 44:
-      case 93:
-      case 125:
-      return ptr
-    }
-  }
-  throw new Error('Could not find end of number')
-}
-
-function indexOfString (s, val, offset) {
-  const i = s.indexOf(val, offset)
-  if (i === -1) throw new Error('Could not parse value')
-  return i
-}
-
-function indexOfKey (keys, val, offset) {
-  const i = keys.indexOf(val, offset)
-  if (i === -1) throw new Error('Unknown key: ' + val)
-  return i
-}
-
-function types (obj) {
-  const t = typeof obj
-  switch (t) {
-    case 'number': return 'number'
-    case 'string': return 'string'
-    case 'boolean': return 'boolean'
-  }
-
-  if (!obj) throw new Error('Cannot infer type')
-
-  if (Array.isArray(obj)) {
-    return [types(obj[0])]
-  }
-
-  const schema = {}
-
-  for (const k of Object.keys(obj)) {
-    schema[k] = types(obj[k])
-  }
-
-  return schema
-}
-
-function isObject (v) {
-  return typeof v === 'object' && !Array.isArray(v)
-}
-
-function bool (v, def) {
-  return typeof v === 'boolean' ? v : def
-}
+/* eslint no-new-func: 0 */
+'use strict'
 
-function stringUnescape (s) {
-  if (s.indexOf('\\') === -1) return s
-  return slowStringUnescape(s)
-}
-
-function slowStringUnescape (s) {
-  return JSON.parse('"' + s + '"') // slow path but unlikely
-}
-
-function compile (schema, opts) {
-  if (!opts) opts = {}
-
-  const unsafe = bool(opts.unsafe, false)
-  const validate = bool(opts.validate, !unsafe)
-  const optional = bool(opts.optional, true)
-  const unescapeStrings = bool(opts.unescapeStrings, true)
-  const validateKeys = bool(opts.validateKeys, !unsafe)
-  const validateLength = bool(opts.validateLength, !unsafe)
-  const buffer = !!opts.buffer
-
-  var tick = 0
-  var incs = 0
-  var absVar = ''
-  var declaredPtr = false
+const genfun = require('generate-function')
 
-  const constants = {endNumber, indexOfString, indexOfKey, endNumberBuffer, endNumberString, stringUnescape}
+module.exports = function (schema) {
   const gen = genfun()
 
-  if (typeof schema !== 'object') throw new Error('Only object or array schemas supported')
+  const constants = {}
 
-  gen('function parse (s, b) {')
-  compileType(null, schema, gen)
-  gen('}')
-
-  return gen.toFunction(constants)
+  var tick = 0
 
-  function inc (n, abs) {
-    if (abs) {
-      absVar = abs
-      incs = n
-    } else {
-      incs += n
-    }
+  function tmp () {
```

Comment:
Maybe use Prettier? This space after the function name looks odd. Is it intentional?

Reply:
> I have used `standard`

> I'll open to change that

I don't force anyone to use `standard` linter. I used to use it. `standard` want that space

Reply:
The space has its fair grounds to distinguish function declaration from function call, but I haven't seen much code using this style, and Prettier looks like becoming a standard de-facto in the frontend/node ecosystem for automatic code formatting for not just JS, but TS, Yaml, Markdown, etc.